server/schedule/operator/create_operator_test.go

```diff
@@ -1078,3 +1080,61 @@ func (suite *createOperatorTestSuite) TestMoveRegionWithoutJointConsensus() {
 		}
 	}
 }
+
+// Ref https://github.com/tikv/pd/issues/5401
+func TestCreateLeaveJointStateOperatorWithoutFitRules(t *testing.T) {
+	re := require.New(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	opts := config.NewTestOptions()
+	cluster := mockcluster.NewCluster(ctx, opts)
+	re.NoError(cluster.SetRules([]*placement.Rule{
+		{
+			GroupID:     "pd",
+			ID:          "default",
+			StartKeyHex: hex.EncodeToString([]byte("")),
+			EndKeyHex:   hex.EncodeToString([]byte("")),
+			Role:        placement.Voter,
+			Count:       1,
+		},
+		{
+			GroupID:     "t1",
+			ID:          "t1",
+			StartKeyHex: hex.EncodeToString([]byte("a")),
+			EndKeyHex:   hex.EncodeToString([]byte("b")),
+			Role:        placement.Voter,
+			Count:       1,
+		},
+		{
+			GroupID:     "t2",
+			ID:          "t2",
+			StartKeyHex: hex.EncodeToString([]byte("b")),
+			EndKeyHex:   hex.EncodeToString([]byte("c")),
+			Role:        placement.Voter,
+			Count:       1,
+		},
+	}))
+	cluster.AddRegionStore(1, 1)
+	cluster.AddRegionStore(2, 1)
+	cluster.AddRegionStore(3, 1)
+	cluster.AddRegionStore(4, 1)
+	originPeers := []*metapb.Peer{
+		{Id: 3, StoreId: 3, Role: metapb.PeerRole_DemotingVoter},
+		{Id: 4, StoreId: 4, Role: metapb.PeerRole_IncomingVoter},
+	}
+
+	region := core.NewRegionInfo(&metapb.Region{Id: 1, Peers: originPeers, StartKey: []byte("a"), EndKey: []byte("c")}, originPeers[0])
```

Comment:
why cannot fit any rules in this case?

Reply:
It crosses two rule(range)

Reply:
I expect the default, is it not normal?

Reply:
It is not normal if crossing multiple ranges, because of RegionFit will not work. The region have to be split first.